### ISSUE TITLE
resource templates do not have any logs when archived

### ISSUE THREAD

**Issue by christopheblin** [argoproj/argo-workflows#6097]
## Summary 

When you have a workflow that uses a resource template and it is archived, you do not see any logs

![Capture d’écran 2021-06-07 à 09 04 25](https://user-images.githubusercontent.com/11044877/120973565-6386ad80-c76f-11eb-8eaa-9e1d59f29fa4.png)

**-> you cannot find the name of the created resource** (especially frustrating when it is another workflow and that you have use workflow-of-workflows with dozens of "children")

"Additional" bug (or maybe related) : **the error message is not recovered in the UI**

instead, there is `Error (exit code 1): interface conversion: error is *exec.Error, not *exec.ExitError` which seems like a bug in Go

![Capture d’écran 2021-06-07 à 09 00 34](https://user-images.githubusercontent.com/11044877/120973241-f5da8180-c76e-11eb-8558-d436d237837e.png)

Additional improvement : **would be great to have a link to the workflow and to the archived workflow when the created resource is a workflow** -> does that need a separate issue ?

---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

**Comment by terrytangyuan** [comment]
I think @alexec is fixing `Error (exit code 1): interface conversion: error is *exec.Error, not *exec.ExitError` in https://github.com/argoproj/argo-workflows/pull/6091

**Comment by christopheblin** [comment]
@terrytangyuan thanks for the info, what about the missing logs : is this related to the error or is this completely an issue on its own ?

**Comment by terrytangyuan** [comment]
I am uncertain at this point yet. I think once #6091 is merged, we will see a much clearer error message.

**Comment by christopheblin** [comment]
ok, I fallback to v3.1.0-rc12 and the logs are also missing on the archive : no way to find the name of the workflow which has been submitted ...

It happens when the pod is in success or error, indifferrently

Note that if I go to the non archived workflow (TTL is 1h), I can see way too much logs :)

![Capture d’écran 2021-06-07 à 18 04 09](https://user-images.githubusercontent.com/11044877/121052556-c6e9fd00-c7ba-11eb-85aa-da11620831df.png)

It would be good if the resource could be logged in debug only and keeping info to the "relevant" evaluations like :

```
time="2021-06-07T15:55:23.163Z" level=info msg="failure condition '{status.phase in [Error Failed]}' evaluated false"
time="2021-06-07T15:55:23.164Z" level=info msg="success condition '{status.phase == [Succeeded]}' evaluated false"
time="2021-06-07T15:55:23.164Z" level=info msg="0/1 success conditions matched"
time="2021-06-07T15:55:23.164Z" level=info msg="Waiting for resource workflow.argoproj.io/dataset-k6h2h in namespace staging-argo resulted in retryable error: Neither success condition nor the failure condition has been matched. Retrying..."
```

Also note that these "temporary" logs (because they disappear in the archive) always end by a "Retrying" (whereas it is clear that something has been evaluated to true to move on ...) -> it seems that the latest logs are not captured correctly

For a resource template, do not you think that argo should automatically create an output with some infos about the created resource ?

at least, the archived logs should tell the id of the created resource (again especially when this is another workflow, it maybe could provide a link to the workflow and to the archive ...)

**Comment by terrytangyuan** [comment]
> Note that if I go to the non archived workflow (TTL is 1h), I can see way too much logs :)

I agree this is too much log. I've adjusted it to debug level in https://github.com/argoproj/argo-workflows/pull/6100.

> For a resource template, do not you think that argo should automatically create an output with some infos about the created resource ?

For this, you can use output parameters + jsonPath/jqFilter to obtain: https://github.com/argoproj/argo-workflows/blob/master/examples/k8s-jobs.yaml#L44-L56

**Comment by christopheblin** [comment]
I tested the outputs.parameters from jsonpath, it worked great and so I documented this behavior in the appropriate section

![Capture d’écran 2021-06-08 à 08 39 48](https://user-images.githubusercontent.com/11044877/121135746-1a992c80-c835-11eb-9f0c-26e9011d1b35.png)

      outputs: {
        parameters: [
          { name: "name", valueFrom: { jsonPath: '{.metadata.name}' } },
          { name: "namespace", valueFrom: { jsonPath: '{.metadata.namespace}' } },
          { name: "uid", valueFrom: { jsonPath: '{.metadata.uid}' } },
        ]
      }

Do you have any idea to generate a direct link to the workflow archive ?

This is basically just the concat of the namespace and the uid, something like :

`<a href="/archived-workflows/staging-argo/7d847603-5f5a-46ab-a832-429fc0488162">see archive</a>`
